addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.milight.internal.protocol;
+
+import org.openhab.binding.milight.internal.MilightThingState;
+
+/**
+ * Implements functionality for the RGB/White bulbs for protocol version 3.
+ * Color temperature and saturation are not supported for this type of bulbs.
+ *
+ * @author David Graeff
+ * @since 2.0
+ *
+ */
+public class MilightV3RGBW extends MilightV3 {
+    protected static final int brightnessLevels = 26;
+
+    public MilightV3RGBW(QueuedSend sendQueue, int zone) {
+        super(10, sendQueue, zone);
+    }
+
+    // We have no real saturation control for RGBW bulbs. If the saturation is under a 50% threshold
+    // we just change to white mode instead.
+    @Override
+    public void setHSB(int hue, int saturation, int brightness, MilightThingState state) {
+        if (saturation < 50) {
+            whiteMode(state);
+        } else {
+            setPower(true, state);
+
+            state.hue360 = hue;
+            byte messageBytes[] = new byte[] { 0x40, make_color(hue), 0x55 };
+            sendQueue.queueRepeatable(uidc(CAT_COLOR_SET), messageBytes);
+
+            if (brightness != -1) {
```

Comment:
Brightness also needs to be applied when saturation is less than 50 (white mode)